### ISSUE TITLE
Launch flow: Race condition causes "Start with free" plan option to disappear

### ISSUE THREAD

**Issue by niranjan-uma-shankar** [Automattic/wp-calypso#40279]
#### Steps to reproduce

1. Login to any account that has an unlaunched site, and also no custom domain or plan attached to it.
2. Go through the site launch flow

**Scenario 1**
In the domain step of the launch flow, select a custom domain --> you will be taken to the Plans step -> hit the Back button -> without much delay, quickly click Skip purchase -> the "Continue with free" option in the launch flow is missing.

**Scenario 2**
In the domain step of the launch flow, select a custom domain --> you will be taken to the Plans step -> hit the Back button -> without much delay, quickly select another custom domain -> select any paid plan -> the checkout page will show two domains in cart.

#### What I expected

In Scenario 1, I expected the start with free option to be visible since I skipped the domain purchase. 
In Scenario 2, I expected only one domain to be in the cart and that should be the most recently selected domain.

#### What happened instead

In Scenario 1, the start with free site option disappeared. 
In Scenario 2, we had two domains in cart.

#### Screenshot / Video

[Missing "Continue with free" option when domain step is skipped](https://user-images.githubusercontent.com/1898325/74688741-55e29880-51b7-11ea-91a9-41948ed8bbf1.gif)
[Two domains in cart gif](https://cloudup.com/cA4gQWyJTxn)

**Comment by niranjan-uma-shankar** [comment]
Relatex fix is in https://github.com/Automattic/wp-calypso/pull/39428

**Comment by lancewillett** [comment]
Hi @niranjan-uma-shankar Does this simply need testing to confirm the issue is fixed?

**Comment by niranjan-uma-shankar** [comment]
> Or is this issue to track down only the race conditions you mentioned (which will need a different pull request)?

This is correct - this issue tracks the race condition, and we need a new PR to fix it.
